### ISSUE TITLE
Regression: Drop-down menus don't register clicks on MacOS in 2.26.0

### ISSUE THREAD

**Issue by clauniel** [gaphor/gaphor#3399]
### Describe the bug
The profile, hamburger, diagram addition, and right click menus can be opened, but clicking on any element of the menu does nothing.

### To Reproduce
1. Click on profile menu
2. Click on any profile other than the one currently selected
3. Observe that the profile is not changed

1. Click on the add diagram button
2. Select any diagram to add
3. Observe that no diagram is added

```
If possible, add the error report from `≡` (top-right menu) → `Tools` → `Error Reports`.
```
Can't do this because clicking `Tools` element closes the menu.

### Expected behavior
The menus should work as they do in Gaphor 2.25.1

### OS
- [ ] Linux (Please put in notes the specific distro)
- [x] macOS 13.6.6
- [ ] Windows

### Version
Version of Gaphor: 2.26.0

**Comment by amolenaar** [comment]
Hi @clauniel, Thank you for your report.

Unfortunately I'm not able to reproduce the issue you describe (Apple MB Air M2, macOS Sanoma 14.5).

Are there any specific error messages you see if you start Gaphor from the terminal (`/Applications/Gaphor.app/Contents/MacOS/gaphor`), except for `Broken accounting of active state` messages)?

**Comment by clauniel** [comment]
Here is the output I get from playing around a bit. I get lots of broken accounting messages, but no messages when interacting with the menus.

This
```
(gaphor:39130): Gdk-CRITICAL **: 13:03:24.983: _gdk_macos_surface_update_fullscreen_state: assertion 'GDK_IS_MACOS_SURFACE (self)' failed
```

Appears when double clicking elements in the model explorer.

Full log:

```
UNSUPPORTED (log once): POSSIBLE ISSUE: unit 1 GLD_TEXTURE_INDEX_2D is unloadable and bound to sampler type (Float) - using zero texture because texture unloadable
gaphor.storage.storage INFO Read 3 elements from file

(gaphor:39130): Gdk-CRITICAL **: 13:02:56.552: _gdk_macos_surface_update_fullscreen_state: assertion 'GDK_IS_MACOS_SURFACE (self)' failed

(gaphor:39130): Gtk-WARNING **: 13:02:56.566: Broken accounting of active state for widget 0x7fb71f609380(GtkLabel)

(gaphor:39130): Gtk-WARNING **: 13:02:56.566: Broken accounting of active state for widget 0x7fb71f609090(GtkBox)

(gaphor:39130): Gtk-WARNING **: 13:02:56.566: Broken accounting of active state for widget 0x7fb71f547d10(GtkToggleButton)

(gaphor:39130): Gtk-WARNING **: 13:02:56.566: Broken accounting of active state for widget 0x7fb71f547b00(GtkMenuButton)

(gaphor:39130): Gtk-WARNING **: 13:02:56.566: Broken accounting of active state for widget 0x7fb71f546f80(GtkBox)

(gaphor:39130): Gtk-WARNING **: 13:02:56.566: Broken accounting of active state for widget 0x7fb71f5468e0(AdwGizmo)

(gaphor:39130): Gtk-WARNING **: 13:02:56.566: Broken accounting of active state for widget 0x7fb71f546580(GtkCenterBox)

(gaphor:39130): Gtk-WARNING **: 13:02:56.566: Broken accounting of active state for widget 0x7fb71f5463f0(GtkWindowHandle)

(gaphor:39130): Gtk-WARNING **: 13:02:56.566: Broken accounting of active state for widget 0x7fb71f5461e0(AdwHeaderBar)

(gaphor:39130): Gtk-WARNING **: 13:02:56.566: Broken accounting of active state for widget 0x7fb71f5459c0(GtkBox)

(gaphor:39130): Gtk-WARNING **: 13:02:56.566: Broken accounting of active state for widget 0x7fb71f545820(GtkWindowHandle)

(gaphor:39130): Gtk-WARNING **: 13:02:56.566: Broken accounting of active state for widget 0x7fb71f545650(GtkRevealer)

(gaphor:39130): Gtk-WARNING **: 13:02:56.566: Broken accounting of active state for widget 0x7fb71f5452d0(AdwToolbarView)

(gaphor:39130): Gtk-WARNING **: 13:02:56.566: Broken accounting of active state for widget 0x7fb71f551e80(GtkPaned)

(gaphor:39130): Gtk-WARNING **: 13:02:56.566: Broken accounting of active state for widget 0x7fb71f552060(AdwToastOverlay)

(gaphor:39130): Gtk-WARNING **: 13:02:56.566: Broken accounting of active state for widget 0x7fb71f552dc0(AdwBreakpointBin)

(gaphor:39130): Gtk-WARNING **: 13:02:56.566: Broken accounting of active state for widget 0x7fb71f552a00(AdwBin)

(gaphor:39130): Gtk-WARNING **: 13:02:56.566: Broken accounting of active state for widget 0x7fb71f552840(AdwDialogHost)

(gaphor:39130): Gtk-WARNING **: 13:02:56.566: Broken accounting of active state for widget 0x7fb71f552330(AdwApplicationWindow)

(gaphor:39130): Gtk-WARNING **: 13:03:04.664: Broken accounting of active state for widget 0x7fb71f609380(GtkLabel)

(gaphor:39130): Gtk-WARNING **: 13:03:04.664: Broken accounting of active state for widget 0x7fb71f609090(GtkBox)

(gaphor:39130): Gtk-WARNING **: 13:03:04.664: Broken accounting of active state for widget 0x7fb71f547d10(GtkToggleButton)

(gaphor:39130): Gtk-WARNING **: 13:03:04.664: Broken accounting of active state for widget 0x7fb71f547b00(GtkMenuButton)

(gaphor:39130): Gtk-WARNING **: 13:03:04.664: Broken accounting of active state for widget 0x7fb71f546f80(GtkBox)

(gaphor:39130): Gtk-WARNING **: 13:03:04.664: Broken accounting of active state for widget 0x7fb71f5468e0(AdwGizmo)

(gaphor:39130): Gtk-WARNING **: 13:03:04.664: Broken accounting of active state for widget 0x7fb71f546580(GtkCenterBox)

(gaphor:39130): Gtk-WARNING **: 13:03:04.664: Broken accounting of active state for widget 0x7fb71f5463f0(GtkWindowHandle)

(gaphor:39130): Gtk-WARNING **: 13:03:04.664: Broken accounting of active state for widget 0x7fb71f5461e0(AdwHeaderBar)

(gaphor:39130): Gtk-WARNING **: 13:03:04.664: Broken accounting of active state for widget 0x7fb71f5459c0(GtkBox)

(gaphor:39130): Gtk-WARNING **: 13:03:04.664: Broken accounting of active state for widget 0x7fb71f545820(GtkWindowHandle)

(gaphor:39130): Gtk-WARNING **: 13:03:04.664: Broken accounting of active state for widget 0x7fb71f545650(GtkRevealer)

(gaphor:39130): Gtk-WARNING **: 13:03:04.664: Broken accounting of active state for widget 0x7fb71f5452d0(AdwToolbarView)

(gaphor:39130): Gtk-WARNING **: 13:03:04.664: Broken accounting of active state for widget 0x7fb71f551e80(GtkPaned)

(gaphor:39130): Gtk-WARNING **: 13:03:04.664: Broken accounting of active state for widget 0x7fb71f552060(AdwToastOverlay)

(gaphor:39130): Gtk-WARNING **: 13:03:04.664: Broken accounting of active state for widget 0x7fb71f552dc0(AdwBreakpointBin)

(gaphor:39130): Gtk-WARNING **: 13:03:04.664: Broken accounting of active state for widget 0x7fb71f552a00(AdwBin)

(gaphor:39130): Gtk-WARNING **: 13:03:04.664: Broken accounting of active state for widget 0x7fb71f552840(AdwDialogHost)

(gaphor:39130): Gtk-WARNING **: 13:03:04.664: Broken accounting of active state for widget 0x7fb71f552330(AdwApplicationWindow)

(gaphor:39130): Gdk-CRITICAL **: 13:03:06.897: _gdk_macos_surface_update_fullscreen_state: assertion 'GDK_IS_MACOS_SURFACE (self)' failed

(gaphor:39130): Gtk-WARNING **: 13:03:06.905: Broken accounting of active state for widget 0x7fb71f54c470(GtkBuiltinIcon)

(gaphor:39130): Gtk-WARNING **: 13:03:06.905: Broken accounting of active state for widget 0x7fb71f54c170(GtkBox)

(gaphor:39130): Gtk-WARNING **: 13:03:06.905: Broken accounting of active state for widget 0x7fb71f54be80(GtkToggleButton)

(gaphor:39130): Gtk-WARNING **: 13:03:06.905: Broken accounting of active state for widget 0x7fb71f54b780(GtkMenuButton)

(gaphor:39130): Gtk-WARNING **: 13:03:06.905: Broken accounting of active state for widget 0x7fb71f54b490(GtkBox)

(gaphor:39130): Gtk-WARNING **: 13:03:06.905: Broken accounting of active state for widget 0x7fb71f54afc0(AdwGizmo)

(gaphor:39130): Gtk-WARNING **: 13:03:06.905: Broken accounting of active state for widget 0x7fb71f54ae30(GtkCenterBox)

(gaphor:39130): Gtk-WARNING **: 13:03:06.905: Broken accounting of active state for widget 0x7fb71f54aca0(GtkWindowHandle)

(gaphor:39130): Gtk-WARNING **: 13:03:06.905: Broken accounting of active state for widget 0x7fb71f54aa90(AdwHeaderBar)

(gaphor:39130): Gtk-WARNING **: 13:03:06.905: Broken accounting of active state for widget 0x7fb71f549b60(GtkBox)

(gaphor:39130): Gtk-WARNING **: 13:03:06.905: Broken accounting of active state for widget 0x7fb71f5499c0(GtkWindowHandle)

(gaphor:39130): Gtk-WARNING **: 13:03:06.905: Broken accounting of active state for widget 0x7fb71f5497f0(GtkRevealer)

(gaphor:39130): Gtk-WARNING **: 13:03:06.905: Broken accounting of active state for widget 0x7fb71f549640(AdwToolbarView)

(gaphor:39130): Gtk-WARNING **: 13:03:06.905: Broken accounting of active state for widget 0x7fb71f5519f0(AdwBin)

(gaphor:39130): Gtk-WARNING **: 13:03:06.905: Broken accounting of active state for widget 0x7fb71f551170(AdwOverlaySplitView)

(gaphor:39130): Gtk-WARNING **: 13:03:06.905: Broken accounting of active state for widget 0x7fb71f551e80(GtkPaned)

(gaphor:39130): Gtk-WARNING **: 13:03:06.905: Broken accounting of active state for widget 0x7fb71f552060(AdwToastOverlay)

(gaphor:39130): Gtk-WARNING **: 13:03:06.905: Broken accounting of active state for widget 0x7fb71f552dc0(AdwBreakpointBin)

(gaphor:39130): Gtk-WARNING **: 13:03:06.905: Broken accounting of active state for widget 0x7fb71f552a00(AdwBin)

(gaphor:39130): Gtk-WARNING **: 13:03:06.905: Broken accounting of active state for widget 0x7fb71f552840(AdwDialogHost)

(gaphor:39130): Gtk-WARNING **: 13:03:06.905: Broken accounting of active state for widget 0x7fb71f552330(AdwApplicationWindow)
gaphor.application INFO Window not closed, abort quit operation

(gaphor:39130): Gdk-CRITICAL **: 13:03:24.983: _gdk_macos_surface_update_fullscreen_state: assertion 'GDK_IS_MACOS_SURFACE (self)' failed

(gaphor:39130): Gdk-CRITICAL **: 13:03:42.584: _gdk_macos_surface_update_fullscreen_state: assertion 'GDK_IS_MACOS_SURFACE (self)' failed
gaphor.application INFO Window not closed, abort quit operation
```

**Comment by amolenaar** [comment]
@clauniel Thanks. Indeed, that statement seems suspicious. Are you running Gaphor in full-screen mode, maximized, or as a normal window?

**Comment by danyeaw** [comment]
@amolenaar is this fixed in #3415?

**Comment by amolenaar** [comment]
@danyeaw No. #3415 just changes how we determine if we should show a popup menu.

**Comment by mathiascode** [comment]
> Could be a related issue: https://gitlab.gnome.org/GNOME/gtk/-/issues/6697.

Looks like that issue. So far it's exclusive to macOS Ventura, which OP is using.

**Comment by tompkins-ct** [comment]
@danyeaw and @amolenaar  I am getting the same issue. Right clicks do not work (no context menu). I'm running in windowed mode with 2.26 and the main branch. 

I have Mac M2 Pro, Sonoma 14.5. Terminal log is the same as above.

**Comment by amolenaar** [comment]
It's fixed in 2.27.0 by 15536691796dbfdebb7e1186f6232fef9988f856.
